### ISSUE TITLE
Rails 5.0 and 5.1 removal released in patch level (?)

### ISSUE THREAD

**Issue by tvdeyen** [activerecord-hackery/ransack#1103]
It seems that the Rails 5.0 and 5.1 specific code removal has been shipped with 2.3.1. Is this correct? If so, was this intentional or a mistake?

Ideally - even if Rails 5.0 and 5.1 are EOL - such removal would have been released as a 3.0 instead.

**Disclaimer**: I read this from the [changelog](https://github.com/activerecord-hackery/ransack/blob/master/CHANGELOG.md#231---2020-01-11) and did not tested this with code. Maybe I am wrong here, but the changelog tells it that way

**Comment by seanfcarroll** [comment]
@tvdeyen thanks for the issue. In order to keep Ransack maintainable older versions are aggressively being dropped. I'm not certain that dropping EOL Rails or Ruby would require a major verion bump (eg to 3). There is a discussion on this [over on Reddit](https://www.reddit.com/r/ruby/comments/7vtceq/when_is_it_okay_to_drop_support_for_ruby_versions/).

It's arguable that dropping a EOL dependency is breaking the API, but we should go forward with a solution that suits the Ransack community. So personally I am happy to do a major release bump if we drop Rails versions, but we should document this.

A possible way forward might be:

- create a v3.0.0 which is essentially 2.3.1
- update the README to indicate that Ransack will be bumped a major release if Rails or Ruby is dropped
- release a 2.3.3 which is essentially 2.3.0

I'm not a huge fan of this, as it is creating work to support older versions, when it is always possible to pin to an [older version of Ransack](https://github.com/activerecord-hackery/ransack/commit/4e148cc0bfc54a95ed64c76e1b92ee43c8f1d1a4#diff-790070a3d6850ff44232a1878aed3f79).

However, if you want to do PR's along these lines I'll be happy to merge them. 

I agree things are a bit messy at the moment, but unfortunately the community contributions are very low. I personally think we should focus on getting Ransack to work with Rails 6 and beyond. So far there are no working PRs to make this happen.

Another important area in Ransack is increasing test coverage and modernize the tests, and replace [machinist](https://github.com/notahat/machinist) which hasn't had a commit since 2013 with fixtures or factories.

**Comment by deivid-rodriguez** [comment]
I was asked about this in the PR that dropped support: https://github.com/activerecord-hackery/ransack/pull/1073.

TL;DR My personal take to this problem is to drop support on minors, which I think it's a good middle ground. That said, I'm good with any approach we finally take here. I fully agree that the focus should be on getting ransack to work on newer stuff, not on supporting old stuff.

As a final note, note that pinning to an older version shouldn't be needed in most situations since bundler should do the right thing by default, namely, only upgrade up to the newest ransack version that still supports your current ruby.

**Comment by seanfcarroll** [comment]
Thanks @deivid-rodriguez 

I also prefer to go with dropping on minors, but if the community feels it should be on majors, we just need this documented in the README via a PR. 

Leaving the floor open here, but if nothing comes forward here in a few days, I'll add a note to the README stating that Ransack will drop dependency versions with minor updates. 

https://semver.org/spec/v2.0.0.html#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api

**Comment by tvdeyen** [comment]
I am by far no semver enthusiast. I also agree that it is fine to drop support in minors. 

But just for the record it seems you dropped support in a patch level. 

Usually I would not have a problem with that either, but lots of people do not pay attention to patch level releases and just merge a PR (from a dependency bot for instance).  

Keep that in mind. 

Thanks for keeping Ransack alive. I maintain a lot of open source as well and I know what a burden it is. 🙏🏻

**Comment by deivid-rodriguez** [comment]
Just to be clear, I agree with you but there's no chance of such PR. If you app is running rails 5.0 for example, a dependency bot will never generate an upgrade to ransack 2.3.1 because it's incompatible with your Rails version.

**Comment by seanfcarroll** [comment]
@tvdeyen are you happy to close this issue out now?
